tests/mock_crust/utils.rs

```diff
@@ -418,6 +418,36 @@ pub fn poll_and_resend_until(
     panic!("poll_and_resend has been called {} times.", MAX_POLL_CALLS);
 }
 
+/// A special advancing pattern to test the ability of resending `CandidateInfo`.
+pub fn poll_and_resend_until_reduced_advance(nodes: &mut [TestNode]) {
```

Comment:
As discussed in mumble, best to not add a duplicate function and make that extra advance clock work in normal poll_and_resend_until.